### PR TITLE
[ Milestone 6 ] Clinic

### PR DESCRIPTION
**Pull request by CakeInTech**
- Created `FK indexes` to enhance the query performance.
- Created a `clinic` schema as provided in the diagram in the `schema_based_on_diagram.sql`.
- Joined tables from many to many relationships.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  invoices_id INT NULL REFERENCES invoices(id) ON DELETE CASCADE,
+  treatment_id INT NULL REFERENCES treatment(id) ON DELETE CASCADE
+)
+
+CREATE TABLE tretments (
+  id BIGSERIAL NOT NULL PRIMARY KEY, 
+  type VARCHAR(40) NOT NULL,
+  name VARCHAR(40) NOT NULL
+)
+
```
Comment (tufoinnkuo10) [inline schema_based_on_diagram.sql:42]: - [x] nice work. please ensure all "join table" for the many-to-many relastionships are created. eg table between `medical_histories` and `treatments`

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+-- Index Clauses 
+CREATE INDEX patient_id ON patients(id);
+CREATE INDEX medical_history_id ON medical_historeis(id);
+CREATE INDEX invoices_id ON invoices(id);
+CREATE INDEX treatment_id ON treatment(id);
```
Comment (tufoinnkuo10) [inline schema_based_on_diagram.sql]: - [x] please ensure that the indexes are created on all FOREIGN KEYS . eg 
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON medical_histories (patient_id); etc

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @MohamedCK 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message


Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (CakeInTech) [reply schema_based_on_diagram.sql:42]: @tufoinnkuo10 Thanks for the reminder.

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (CakeInTech) [reply schema_based_on_diagram.sql]: Thanks for the reminder. 😄

**Review by Deepakdanger — APPROVED** [review]
Hi @TEAM, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**